provider/cmd/pulumi-gen-eks/main.go

```diff
@@ -444,9 +462,20 @@ func generateSchema() schema.PackageSpec {
 					"storageClasses": {
 						TypeSpec: schema.TypeSpec{
 							OneOf: []schema.TypeSpec{
-								{Type: "string"}, // TODO: EBSVolumeType enum "io1" | "gp2" | "sc1" | "st1"
-								{Type: "object", AdditionalProperties: &schema.TypeSpec{Ref: "#/types/eks:index:StorageClass"}},
+								{
+									Type:  "string", // TODO: EBSVolumeType enum "io1" | "gp2" | "sc1" | "st1"
+									Plain: true,
+								},
+								{
+									Type: "object",
+									AdditionalProperties: &schema.TypeSpec{
+										Ref:   "#/types/eks:index:StorageClass",
+										Plain: true,
+									},
+									Plain: true,
+								},
 							},
+							Plain: true,
 						},
```

Comment:
This is plain in the implementation:

https://github.com/pulumi/pulumi-eks/blob/c6f1fce98fa2683abe7ffbad7f166da99928fb83/nodejs/eks/cluster.ts#L1137-L1145